src/reducers/formReducer.js

```diff
@@ -2,12 +2,34 @@ import * as types from 'actions/actionTypes';
 import createReducer from './createReducer';
 
 const initialState = {
-  categories: []
+  categories: [],
+  byId: {},
+  isFetching: true,
+  error: ''
 };
 
 const actionHandlers = {
+  [types.FETCH_FORM_CATEGORIES]: state => {
+    state.isFetching = true;
+  },
   [types.FETCH_FORM_CATEGORIES_SUCCESS]: (state, { payload }) => {
     state.categories = payload.formCategories;
+    state.isFetching = false;
+  },
+  [types.FETCH_FORM_CATEGORIES_ERROR]: (state, { payload }) => {
+    state.error = payload.data.message;
+    state.isFetching = false;
+  },
+  [types.FETCH_FORMS]: state => {
+    state.isFetching = true;
+  },
+  [types.FETCH_FORMS_SUCCESS]: (state, { payload }) => {
+    state.byId[payload.category] = payload.forms;
+    state.isFetching = false;
+  },
+  [types.FETCH_FORMS_ERROR]: (state, { payload }) => {
+    state.error = payload.data.message;
+    state.isFetching = false;
```

Comment:
You could omit al this boilerplate if you rely on the statusReducer to manage action's statuses. I really think this https://github.com/rootstrap/react-redux-base/blob/master/src/reducers/statusReducer.js would help a ton (plus the two hooks that are on the hooks folder 😉 )

Reply:
Alright, going to take a look to reduce the boilerplate :)